pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -293,8 +293,9 @@ private boolean createDictionaryForColumn(ColumnIndexCreationInfo info, SegmentG
 
     FieldIndexConfigs fieldIndexConfigs = config.getIndexConfigsByColName().get(column);
     if (DictionaryIndexType.ignoreDictionaryOverride(config.isOptimizeDictionary(),
-        config.isOptimizeDictionaryForMetrics(), config.getNoDictionarySizeRatioThreshold(), spec, fieldIndexConfigs,
-        info.getDistinctValueCount(), info.getTotalNumberOfEntries())) {
+        config.isOptimizeDictionaryForMetrics(), config.getNoDictionarySizeRatioThreshold(),
```

Comment:
This list has become quite long. Suggest adding a new config class for dictionary tuning related configs

Reply:
After looking at it I feel the config and builder class for just four configs is a lot, since I don't think this would be a commonly edited part of the code and is only used in two places - wdyt? If you prefer it I'm still open to making the change

Reply:
Having 4 configs on the same topic passed in separately is a little bit ugly, but we can address it separately